README.md

```diff
@@ -143,4 +143,58 @@ In case you experience problems concerning the submodule checkout, i.e. the clon
 
 # Documentation
 
-TODO
+### Structure
+
+- VulkanLaunchpad.h/VulkanLaunchpad.cpp: main functionality of the framework.
+- Camera.h/Camera.cpp: implementation of an orbit/arcball camera.
+- external: contains the libraries glfw, glm and glslang as git submodules, as well as tinyobjloader and gli.
+
+### Naming Conventions
+
+Function names are in lowerCamelCase starting with the prefix `vkl`. Structs and classes are in UpperCamelCase starting with the prefix `Vkl`. Member variables are mCamelCase, function parameters are lowercase_with_underscores.
+
+### Functionality
+
+The framework needs to be initinilized by calling `vklInitFramework`. Subsequently, it needs to be destroyed via `vklDestroyFramework`. Required extensions can be queried by `vklGetRequiredInstanceExtensions`. Consistent with the Vulkan API, custom configuration structs should best be zero-initialized, although they do not require to expilictly set their instance's type:
+
+    VklSwapchainConfig swapchain_config = {};
+    
+#### General
+
+The basic graphics pipeline created by the framework can be retrieved using `vklGetBasicPipeline`. To set up a custom graphics pipeline you can use `vkl(Create/Destroy)GraphicsPipeline`.
+
+The Launchpad also provides functionality needed during a typical render loop:
+
+    vklWaitForNextSwapchainImage();
+	vklStartRecordingCommands();
+		
+    // Your commands here
+
+	vklEndRecordingCommands();
+	vklPresentCurrentSwapchainImage();
+
+#### Buffer Management
+
+The Vulkan Launchpad can help with buffer management by allocating different types of memory and transfering data to fill the buffer. Two main functions for creating a buffer are:
+
+- `vklCreateHostCoherentBufferWithBackingMemory` : Creates a new `VkBuffer` and allocates memory both on the device and in a region of so called "host coherent" memory, which is also accessible from the CPU. 
+- `vklCreateDeviceLocalBufferWithBackingMemory` : Creates a new `VkBuffer` and allocates memory in a region of so called "device-local" memory, which is a region that is not accessible from the CPU, but which is faster to access and transfer on the device.
+
+Both need to be destroyed by calling `vklDestroy(HostCoherent|DeviceLocal)BufferAndItsBackingMemory`. The host coherent buffer can be filled by calling `vklCopyDataIntoHostCoherentBuffer` or alternatively, the data can already be supplied during creation using `vklCreateHostCoherentBufferAndUploadData`.
+
```

Comment:
I think we need a new section for Descriptor Sets, introducing the `vklBindDescriptorSetToPipeline` function, which allows on to bind a `VkDescriptorSet` to the pipeline layout of a given `VkPipeline`.

Reply:
Yeah, true.. but in the tutorial, we mostly require manual descriptor set handling. So, maybe add a quick note about `vklBindDescriptorSetToPipeline` to the "Graphics Pipelines" section?!